tensorboard/plugins/core/core_plugin.py

```diff
@@ -88,14 +91,47 @@ def get_plugin_apps(self):
         '/histograms': self._redirect_to_index,
         '/images': self._redirect_to_index,
     }
-    if self._assets_zip_provider:
-      with self._assets_zip_provider() as fp:
-        with zipfile.ZipFile(fp) as zip_:
-          for path in zip_.namelist():
-            gzipped_asset_bytes = _gzip(zip_.read(path))
-            apps['/' + path] = functools.partial(
+    apps.update(self.get_resource_apps())
+    return apps
+
+  def get_resource_apps(self):
+    apps = {}
+    if not self._assets_zip_provider:
+      return apps
+
+    with self._assets_zip_provider() as fp:
+      with zipfile.ZipFile(fp) as zip_:
+        # First collect the shasums that will later be used. Since hashes are
+        # in the hex, we convert it to base64 for CSP usage.
+        # TODO(stephanwlee): since index.html creates dynamic plugin and imports
+        # their entry point (es_module_path), the CSP should include shasum
+        # of those, too.
+        shasums = {}
+        for path in zip_.namelist():
+          if not path.startswith(SHASUM_DIR):
+            continue
+          html_file_name = os.path.splitext(os.path.basename(path))[0]
```

Comment:
Any reason we're calling `basename()` here?  It seems like then we have a risk of colliding if two files in different directories have the same basename?

(Same applies to where we call it down below to compute `filename`.)